### PR TITLE
added express-rate-limiter

### PR DESCRIPTION
**Pull request by marcschwaiger**
Using express-rate-limiter we can block e. g. brute force attacks on admin login
(Note that using proxys bypasses this check. Consider using software to block proxy requests)

### CODE REVIEW

**Comment by marcschwaiger** [comment]
Needs to be reviewed @94726

**Comment by marcschwaiger** [comment]
Had to remove --silent in order to see the error I had. I recommend keeping the output from npm ci

File: src/app.ts [hunk 1]
```diff
@@ -2,13 +2,36 @@ import compression from 'compression';
...
+
+const adminLoginLimiter = rateLimit({
+  windowMs: 15 * 60 * 1000, // 15 minutes
+  max: 5,
+  standardHeaders: true,
+  legacyHeaders: false,
+});
+
+// Apply the rate limiting to admin login requests
+app.use("/admin/login", adminLoginLimiter);
```
Comment (94726) [inline src/app.ts]: Maybe put the middleware use directly into router/index.ts similar to other middleware uses

**Review by 94726 — CHANGES_REQUESTED** [review]

File: src/app.ts (hunk 1 above)
Comment (marcschwaiger) [reply src/app.ts]: Resolved in [moved rate limiter config to its own class](https://github.com/LeoTuet/backend-we-want-to-sleep/pull/29/commits/c53c61c4be3a91d21fc66df9391bf5277657553b)[](https://github.com/LeoTuet/backend-we-want-to-sleep/pull/29/commits/c53c61c4be3a91d21fc66df9391bf5277657553b) and [updated index.ts to use rate limiter](https://github.com/LeoTuet/backend-we-want-to-sleep/pull/29/commits/4c9256109e7e34369b7562afe369a933b01367eb)

**Comment by marcschwaiger** [comment]
In order for this to work on the productive system, the [configuration file patch](https://github.com/LeoTuet/deployment-we-want-to-sleep/pull/4) must be applied

**Review by 94726 — APPROVED** [review]

**Pull request merged**